src/Vector.js

```diff
@@ -0,0 +1,115 @@
+/* Convenience class to store a vector and its associated flags */
+class Vector
+{
+	constructor(x, y)
+	{
+		this.components = [x, y];
+	}
+	
+	/**
+	 *  Creates one or more Vectors based on the string passed in.
+	 *  Returns an array of Vectors containing all Vectors described in the string.
+	 *
+	 *  If an invalidly formatted string is passed in, an error is logged to the
+	 *  console, but execution continues (and all validly formatted Vectors are
+	 *  output normally).
+	 */
+	static Create(str)
+	{
+		/* helper function */
+		function buildComponent(compStr, globalStr)
+		{
+			var comp = new Component(0, 1, false, false, false);
+			comp.length = Number(compStr.match(/(-?\d+)/g)[0]);
+			
+			/* Check for jump on first component or end */
+			if (compStr.includes("j") || globalStr.includes("j"))
+			{
+				comp.jump = true;
+			}
+			
+			/* Check for hop on first component or end */
+			if (compStr.includes("h") || globalStr.includes("h"))
+			{
+				comp.hop = true;
+			}
+			
+			/* Check for promote on first component or end */
+			if (compStr.includes("p") || globalStr.includes("p"))
+			{
+				comp.promote = true;
+			}
+			
+			/* Check for finite repetition on end */
+			var repetition = globalStr.match(/{(\d+)}/g);
+			if (repetition)
+			{
+				comp.maxRep = Number(repetition[0].slice(2, -1));
+			}
+			
+			/* Check for infinite repetition */
+			if(compStr.includes("+") || globalStr.includes("+"))
+			{
+				comp.maxRep = 100; /* arbitrarily large */
+			}
+	
+			/* Check for finite specific repeition */
+			repetition = compStr.match(/{(\d+)}/);
+			if (repetition)
+			{
+				comp.maxRep = Number(repetition[1]);
+			}
+			
+			/* Check for directional or zero length */
+			if (compStr.includes("d") || globalStr.includes("d") || comp.length == 0)
+			{
+				return [comp];
+			}
+			
+			var reverseComp = new Component(comp.length * -1, comp.maxRep, 
+				comp.jump, comp.hop, comp.promote);
+				
+			return [comp, reverseComp];
+		}
+		
+		/* Make all possible vectors */
+		var toReturn = [];
+		var substrings = str.split(";");
+		for (var i = 0; i < substrings.length; i++)
+		{
+			var vectorString = substrings[i].trim();
+			if (vectorString == "")
+			{
+				/* silent error - common if vector list ended with semicolon */
```

Comment:
If it doesn't throw an error, it isn't technically an error.
```suggestion
				/* Ignore empty strings - common if vector list ended with semicolon */
```